### PR TITLE
Deprecate push notifications methods

### PR DESCRIPTION
**Pull request by luismfonseca**
#### Why is the change necessary?

We want to recommend everyone to use Pusher Beams as this feature is being removed from Channels.

----

CC @pusher/mobile

### CODE REVIEW

File: Library/src/main/java/com/pusher/android/PusherAndroid.java [hunk 1]
```diff
@@ -109,6 +109,11 @@ public PresenceChannel getPresenceChannel(String channelName) {
         return this.pusher.getPresenceChannel(channelName);
     }
 
+    /**
+     * @return the client for push notifications
+     * @deprecated this feature is being decommissioned, please use Pusher Beams instead
```
Comment (hosamaly) [inline Library/src/main/java/com/pusher/android/PusherAndroid.java]: I think that we should clarify that we'll remove this method in the next version.

File: Library/src/main/java/com/pusher/android/notifications/PushNotificationRegistration.java [hunk 2]
```diff
@@ -59,14 +59,20 @@ public PushNotificationRegistration(
         this.manifestValidator = manifestValidator;
     }
 
+    /**
+     * @deprecated this feature is being decommissioned, please use Pusher Beams instead
```
Comment (hosamaly) [inline Library/src/main/java/com/pusher/android/notifications/PushNotificationRegistration.java]: How about adding a link to Beams' home page?

File: Library/src/main/java/com/pusher/android/notifications/PushNotificationRegistration.java [hunk 3]
```diff
@@ -59,14 +59,20 @@ public PushNotificationRegistration(
...
         registerGCM(context, defaultSenderId, null);
     }
 
-    /*
-    Starts a PusherRegistrationIntentService, which handles token receipts and updates from
-    GCM
+    /**
+     * Starts a PusherRegistrationIntentService, which handles token receipts and updates from GCM
+     *
+     * @deprecated this feature is being decommissioned, please use Pusher Beams instead
```
Comment (hosamaly) [inline Library/src/main/java/com/pusher/android/notifications/PushNotificationRegistration.java]: The word "this feature" can be misunderstood. Can we be more specific?

File: Library/src/main/java/com/pusher/android/PusherAndroid.java (hunk 1 above)
Comment (luismfonseca) [reply Library/src/main/java/com/pusher/android/PusherAndroid.java]: Aren't all methods marked as deprecated a signal that they are going to be removed eventually?

Also, I don't know when we are releasing the next version of this library as it doesn't make sense to exist without the push notifications functionality -- you can just use the `pusher-websocket-java`.

File: Library/src/main/java/com/pusher/android/notifications/PushNotificationRegistration.java (hunk 2 above)
Comment (luismfonseca) [reply Library/src/main/java/com/pusher/android/notifications/PushNotificationRegistration.java]: I was thinking about it. I was also wondering about the migration guide, but I was worried that it's going to go away in a few months and no one will update the library to fix a dead-link.

File: Library/src/main/java/com/pusher/android/notifications/PushNotificationRegistration.java (hunk 3 above)
Comment (luismfonseca) [reply Library/src/main/java/com/pusher/android/notifications/PushNotificationRegistration.java]: What do you think of:
> Push Notifications no longer supported on this SDK. Please use Pusher Beams: www.pusher.com/beams

Comment (hosamaly) [reply Library/src/main/java/com/pusher/android/notifications/PushNotificationRegistration.java]: 👍 
("are no longer")

File: Library/src/main/java/com/pusher/android/notifications/PushNotificationRegistration.java (hunk 2 above)
Comment (hosamaly) [reply Library/src/main/java/com/pusher/android/notifications/PushNotificationRegistration.java]: I think that the link to the migration guide is a good one. I wouldn't worry about the dead link.

I also think that such pages should remain online, even if the content is replaced with "This is an old guide for a discontinued product."

File: Library/src/main/java/com/pusher/android/PusherAndroid.java (hunk 1 above)
Comment (hosamaly) [reply Library/src/main/java/com/pusher/android/PusherAndroid.java]: In the Java world, some "deprecated" APIs [are still there](https://docs.oracle.com/javase/8/docs/api/deprecated-list.html) after 20 years!

But I agree with you about releasing the next version. Although I think that the work on end-to-end encryption will require a new release.

**Review by hosamaly — APPROVED** [review]
There is nothing critical in the remaining comments.

File: Library/src/main/java/com/pusher/android/notifications/PushNotificationRegistration.java (hunk 3 above)
Comment (kn100) [reply Library/src/main/java/com/pusher/android/notifications/PushNotificationRegistration.java]: Push Notifications are no longer supported *in* this SDK. Please use Pusher Beams, our new Push Notifications product: www.pusher.com/beams

is my suggestion. Mainly the use of the word **in** though.

**Review by kn100 — APPROVED** [review]
Tres bien

**Pull request merged**